### ISSUE TITLE
Error using pex + virtualenv >=20.0.0 + python 2.7

### ISSUE THREAD

**Issue by borgstrom** [pantsbuild/pex#992]
👋 

I know that Python 2.7 is "dead", but we've run into an issue that is preventing us from upgrading virtualenv and we still have a handful of projects that are yet to migrate to python 3.

This is reproducible on macOS.  I haven't tried other OSes.

Steps to reproduce.

Make sure you have the latest virtualenv (although any release after 20.0.0 should exhibit this behavior).

```
PIP_REQUIRE_VIRTUALENV=false /usr/local/bin/pip install virtualenv --upgrade
```

Create a new 2.7 venv.

```
virtualenv venv --python python2.7
```

Install pex.

```
./venv/bin/pip install pex
```

Try to create a pex

```
./venv/bin/pex -o test requests
```

This results in the following traceback:

```
Traceback (most recent call last):
  File "/Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6/.bootstrap/pex/pex.py", line 401, in execute
    exit_code = self._wrap_coverage(self._wrap_profiling, self._execute)
  File "/Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6/.bootstrap/pex/pex.py", line 333, in _wrap_coverage
    return runner(*args)
  File "/Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6/.bootstrap/pex/pex.py", line 364, in _wrap_profiling
    return runner(*args)
  File "/Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6/.bootstrap/pex/pex.py", line 452, in _execute
    return self.execute_entry(self._pex_info.entry_point)
  File "/Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6/.bootstrap/pex/pex.py", line 549, in execute_entry
    return runner(entry_point)
  File "/Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6/.bootstrap/pex/pex.py", line 553, in execute_module
    cls.demote_bootstrap()
  File "/Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6/.bootstrap/pex/pex.py", line 474, in demote_bootstrap
    for module in bootstrap.demote():
  File "/Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6/.bootstrap/pex/bootstrap.py", line 42, in demote
    import sys  # Grab a hold of `sys` early since we'll be un-importing our module in this process.
  File "/Users/eborgstrom/scratch/pex-repro/venv/lib/python2.7/site-packages/_virtualenv.py", line 97, in find_module
    if fullname in _DISTUTILS_PATCH:
TypeError: argument of type 'NoneType' is not iterable
pid: 77981 -> /Users/eborgstrom/scratch/pex-repro/venv/bin/python /Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6 --disable-pip-version-check --isolated --no-python-version-warning -q --cache-dir /Users/eborgstrom/.pex download --dest /var/folders/yp/t0j_kl_d1mj2_8bvs8_w_wg80000gn/T/tmpg3cvzt/cp27-cp27m --header Cache-Control:max-age=3600 --retries 5 --timeout 15 requests raised Executing /Users/eborgstrom/scratch/pex-repro/venv/bin/python /Users/eborgstrom/.pex/pip.pex/6bd62cbe8c9f1ae82b35f976967538908799cbd6 --disable-pip-version-check --isolated --no-python-version-warning -q --cache-dir /Users/eborgstrom/.pex download --dest /var/folders/yp/t0j_kl_d1mj2_8bvs8_w_wg80000gn/T/tmpg3cvzt/cp27-cp27m --header Cache-Control:max-age=3600 --retries 5 --timeout 15 requests failed with 1
```

When this error occurs `fullname` is "pex.sys" and `_DISTUTILS_PATCH` is `None`.  Up until that point `_DISTUTILS_PATCH` is the default of `_DISTUTILS_PATCH = "distutils.dist", "setuptools.dist"` so I'm guessing that pex is monkey patching that value.

This does not occur when using virtualenv 20+ and python 3, nor does it occur using virtualenv 16.7.9 and python 2.

I tried to search through the pex codebase for what might be causing this but I couldn't find anything. Do you have any guidance on what might be the cause? I'd happily work on a fix if there's a way to make this work with python 2.7.

Thanks.

**Comment by parafiend** [comment]
I'm having the same issue on Ubuntu. It looks like virtualenv switched to using import hooks for patching distutils and setuptools in pypa/virtualenv#1688, merged into virtualenv 20.0.8+, so I'm guessing something's getting broken there by the unimport in Bootstrap.demote().

**Comment by jsirois** [comment]
I apologize for the huge delay here. I just hit this with Pex's own tests, which cover 2.7. @parafiend you were pretty much right. It was not Bootstrap.demote(), but it was the sys.path scrubbing process here:
https://github.com/pantsbuild/pex/blob/4aecc86263ccf8d27630ef761335591ab5100759/pex/pex.py#L230-L251

I'll be putting in a special case for the "_virtualenv" module to not be dropped, which breaks the `_VirtualenvImporter` you found.

**Comment by jsirois** [comment]
This fix is needed for #1871.
